libcaf_core/caf/typed_actor.hpp

```diff
@@ -249,11 +260,59 @@ class typed_actor : detail::comparable<typed_actor<Sigs...>>,
     return ptr_.release();
   }
 
-  typed_actor(actor_control_block* ptr) : ptr_(ptr) {
+  strong_actor_ptr ptr_;
+};
+
+/// Identifies a statically typed actor.
+/// @tparam Ts Signatures of all accepted messages.
+/// @note This is a specialization for backwards compatibility with pre v1.0
+/// releases. Please use the trait based implementation.
+template <class T1, class T2, class... Ts>
+class typed_actor<T1, T2, Ts...>
+  : public typed_actor<statically_typed<T1, T2, Ts...>> {
+public:
+  template <class...>
+  friend class typed_actor;
+
+  using super = typed_actor<statically_typed<T1, T2, Ts...>>;
+
+  using behavior_type = typed_behavior<T1, T2, Ts...>;
+
+  /// The default, event-based type for implementing this messaging interface.
+  using impl = typed_event_based_actor<T1, T2, Ts...>;
+
+  /// Identifies pointers to instances of this kind of actor.
+  using pointer = impl*;
+
+  /// A class type suitable as base type class-based implementations.
+  using base = impl;
+
+  /// The default, event-based type for implementing this messaging interface as
+  /// a stateful actor.
+  template <class State>
+  using stateful_impl = stateful_actor<State, impl>;
+
+  /// Convenience alias for `stateful_impl<State>*`.
+  template <class State>
+  using stateful_pointer = stateful_impl<State>*;
+
+  /// A view to an actor that implements this messaging interface without
+  /// knowledge of the actual type.
+  using pointer_view = typed_actor_pointer<T1, T2, Ts...>;
+
+  using super::super;
+
+  typed_actor(const super& other) : super(other) {
+    // nop
+  }
+  typed_actor(super&& other) : super(std::move(other)) {
     // nop
   }
```

Comment:
```suggestion
  typed_actor(const super& other) = default;
  
  typed_actor(super&& other) = default;
```

Reply:
Turns out you can't default these, since they are not special members.

Reply:
Can't we just `using super::super;` here?

Reply:
Actually, `using super::super` worked. I've had it like that in the first implementation and could have sworn that the compiler complained stating that you can't import the copy ctor.

Reply:
Seems like it can't use the copy ctor after all. When I removed the two ctor overloads with `super` from the `typed_behavior`, I got this error:
```
/home/samir/projects/actor-framework/libcaf_core/caf/typed_behavior.hpp:147:7: note: candidate constructor (the implicit move constructor) not viable: cannot bind base class object of type 'typed_behavior<statically_typed<result<void> (put_atom, int), result<int> (get_atom)>>' to derived class reference 'typed_behavior<result<void> (put_atom, int), result<int> (get_atom)> &&' for 1st argument
/home/samir/projects/actor-framework/libcaf_core/caf/typed_behavior.hpp:194:3: note: candidate inherited constructor not viable: no known conversion from 'typed_behavior<statically_typed<result<void> (put_atom, int), result<int> (get_atom)>>' to 'partial_behavior_init_t' for 1st argument
  typed_behavior(partial_behavior_init_t, Ts... xs)
  ^
/home/samir/projects/actor-framework/libcaf_core/caf/typed_behavior.hpp:277:16: note: constructor from base class 'typed_behavior<caf::statically_typed<caf::result<void> (caf::put_atom, int), caf::result<int> (caf::get_atom)>>' inherited here
  using super::super;
               ^
```
I get this error when I remove the `super` overload for typed behavior. Not sure why this doesn't affect